components/fxa-client/src/events.rs

```diff
@@ -0,0 +1,50 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use crate::FxaState;
+use crate::Device;
+
+/// FxA event handler trait
+pub trait FxaEventHandler {
+    fn on_event(&self, event: FxaEvent);
+    fn on_command(&self, command: FxaCommand);
+}
+
+/// FxA event that the app may want to respond to
+pub enum FxaEvent {
+    /// The FxA client state changed
+    StateChanged {
+        state: FxaState,
+    },
+    /// The account itself was deleted
+    AccountDeleted,
+    /// Error when trying to migrate old account data
+    AccountMigrationFailed {
+        // TODO: what fields would be useful here
+    },
+    /// The account's profile was updated
+    ProfileUpdated,
+    /// The list of devices changed
+    DevicesChanged {
+        /// Device record for this client
+        client_devices: Device,
+        /// Device record for all other known clients
+        other_devices: Vec<Device>,
+    },
+    // I don't think we need these:
+    // DeviceConnected
+    // DeviceDisconnected
+    // DeviceCommandIncoming
+    //
+    // I also don't think we should implement the `SyncStatusObserver` functionality.  If needed,
+    // we should add that to SyncManager
+}
+
+/// Received FxA command
+pub enum FxaCommand {
```

Comment:
It's difficult to know where this is heading, but I wonder if we should support multiple event receivers? I don't see a great reason to insist on exactly 1 receiver but then forcing the consuming app to handle the multi-receiver case. Eg, desktop wants many receivers (but it's probably true that desktop would make the adapting to multiple receivers trivial via its observer models)

Why isn't a command just part of the `FxaEvent` enum, meaning you just need an `on_event()` handler which also handles `on_command()`? Although I guess one reason might be if you did handle multiple event handlers - you probably *would* want to insist on a single command handler?